stdlib/sql/to.go

```diff
@@ -284,37 +298,91 @@ func getTranslationFunc(driverName string) (func() translationFunc, error) {
 	}
 }
 
+// bareIdent returns the string it accepts.
+//
+// At this time, this is a stand-in for the quote/escape strategy needed for hbd
+// (which needs testing before we can figure out the correct impl).
+func bareIdent(s string) string {
+	return s
+}
+
+func getQuoteIdentFunc(driverName string) (quoteIdentFunc, error) {
+	switch driverName {
+	case "sqlite3":
+		return doubleQuote, nil
+	case "postgres", "sqlmock":
+		return PostgresQuoteIdent, nil
+	case "vertica", "vertigo":
+		return doubleQuote, nil
+	case "mysql":
+		return MysqlQuoteIdent, nil
+	case "snowflake":
+		// n.b. snowflake automatically UPPERCASES identifiers when they are
+		// specified as bare words in queries (and DDL). Case is preserved when
+		// identifiers are quoted.
+		// Therefore, snowflake users may see breakage with this quoting behavior
+		// since they will now need to explicitly match the case of the
+		// identifiers already defined in schema.
+		return doubleQuote, nil
+	case "mssql", "sqlserver":
+		return doubleQuote, nil
+	case "awsathena": // read-only support for AWS Athena (see awsathena.go)
+		return nil, errors.Newf(codes.Invalid, "writing is not supported for %s", driverName)
+	case "bigquery":
+		// BigQuery offers 2 dialects, "legacy" and "standard."
+		// For the "legacy" dialect, it seems to use MS-style quoting with
+		// square brackets.
+		// The "standard" dialect (which is the default) seems to use backticks
+		// in the style of MySQL (which makes sense for Google from a product
+		// standpoint since their "Cloud SQL" product also speaks this dialect).
+		return MysqlQuoteIdent, nil
+	case "hdb":
+		// FIXME(onelson): unable to run any tests against hdb so far.
+		//  Until we can observe the engine in action, leave idents as-is.
+		return bareIdent, nil
+	default:
+		return nil, errors.Newf(codes.Internal, "invalid driverName: %s", driverName)
+	}
+}
+
 func supportsTx(driverName string) bool {
 	return driverName != "sqlmock" && driverName != "awsathena"
 }
 
 func CreateInsertComponents(t *ToSQLTransformation, tbl flux.Table) (colNames []string, valStringArray [][]string, valArgsArray [][]interface{}, err error) {
 	cols := tbl.Cols()
 	batchSize := correctBatchSize(t.spec.Spec.BatchSize, len(cols))
+	driverName := t.spec.Spec.DriverName
+
+	quoteIdent, err := getQuoteIdentFunc(driverName)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	// the following allows driver-specific type errors (of which there can be MANY) to be returned, rather than the default of invalid type
+	translateColumn, err := getTranslationFunc(driverName)
+	if err != nil {
+		return nil, nil, nil, err
+	}
 
 	labels := make(map[string]idxType, len(cols))
 	var questionMarks, newSQLTableCols []string
 	for i, col := range cols {
-		labels[col.Label] = idxType{Idx: i, Type: col.Type}
+
+		quotedColumName := quoteIdent(col.Label)
+		labels[quotedColumName] = idxType{Idx: i, Type: col.Type}
```

Comment:
Using the quoted form of the identifiers here is what caused the existing tests to require an update. I think ideally, the quoting would happen as close to the string formatting as possible, but I elected to quote earlier (here) to make the mapping here less error prone.

This is to say, I found myself unsure when I should be using quoted or unquoted labels in the rest of the func body. Changing to an "always quoted" approach simplified the flow (for me).